daxa/mission/erosita.py

```diff
@@ -0,0 +1,703 @@
+                    def callback(download_conf: Any):
+                        """
+                        Callback function for the apply_async pool method, gets called when a download task finishes
+                        without error.
+
+                        :param Any download_conf: The Null value confirming the operation is over.
+                        """
+                        nonlocal download_prog  # The progress bar will need updating
+                        download_prog.update(1)
+
+                    # The error callback function is what happens when an exception is thrown during a _download_call
+                    def err_callback(err):
+                        """
+                        The callback function for errors that occur inside a download task running in the pool.
+
+                        :param err: An error that occurred inside a task.
+                        """
+                        nonlocal raised_errors
+                        nonlocal download_prog
+
+                        if err is not None:
+                            # Rather than throwing an error straight away I append them all to a list for later.
+                            raised_errors.append(err)
+                        download_prog.update(1)
+
+                    # Again nested for loop through ObsIDs and instruments
+                    for field in fields_to_be_downloaded:
+                        # Add each download task to the pool
+                        pool.apply_async(self._download_call,
+                                            kwds={'field': field},
+                                            error_callback=err_callback, callback=callback)
+                    pool.close()  # No more tasks can be added to the pool
+                    pool.join()  # Joins the pool, the code will only move on once the pool is empty.
+
+                # Raise all the download errors at once, if there are any
+                if len(raised_errors) != 0:
+                    raise DAXADownloadError(str(raised_errors))
+
+            else:
+                raise ValueError("The value of NUM_CORES must be greater than or equal to 1.")
+
+            # Rearranging the obs_id eventlists into the directory format DAXA expects
+            self._directory_formatting()
+
+            # Only doing the instrument filtering step if not all the instruments have been chosen
+            if len(self.chosen_instruments) != 7:
+                # Getting all the path for each eventlist corresponding to an obs_id for the _inst_filtering function later
+                fits_paths = [self._get_evlist_path_from_obs(obs=o) for o in self.filtered_obs_ids]
+
+                # Filtering out any events from the raw data that arent from the selected instruments
+                if num_cores == 1:
+                    with tqdm(total=len(self), desc="Selecting EventLists from {}".format(self.chosen_instruments)) as download_prog:
```

Comment:
Setting up the progress bar with the variable name 'download_prog' doesn't really make sense here as you aren't downloading.

Reply:
yup, changed the name